### PR TITLE
[#284] Revert the use of level count to denote time intervals

### PR DESCRIPTION
**Pull request by sras**
## Description

Reverts the use of Levels for denoting time intervals. 

## Related issue(s)

Resolves: part of #284

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

[//]: # (Add more docs here if you have them in the repository)
- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

- [x] My commits comply with [the following policy](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: haskell/test/Test/Ligo/BaseDAO/Common.hs [hunk 1]
```diff
@@ -42,6 +47,14 @@ import Test.Ligo.BaseDAO.Proposal.Config (ConfigDesc, fillConfig)
 
 type OriginateFn m = QuorumThreshold -> m DaoOriginateData
 
+type Seconds = Time Second
+
+doubleTime :: Time Second -> Time Second
+doubleTime c = timeMul 2 c
```
Comment (pasqu4le) [inline haskell/test/Test/Ligo/BaseDAO/Common.hs:53]: Nitpick, but since we use `+:+` below we could also use `*:*` here

**Review by pasqu4le — APPROVED** [review]
Only nitpicks, but LGTM

**Comment by pasqu4le** [comment]
Ah @sras one last thing: the first commit's message contains one too many `changes`, I believe.

File: haskell/test/Test/Ligo/BaseDAO/Common.hs (hunk 1 above)
Comment (sras) [reply haskell/test/Test/Ligo/BaseDAO/Common.hs]: Yes, changed.

File: src/proposal/quorum_threshold.mligo [hunk 2]
```diff
@@ -49,8 +49,7 @@ let bound_qt (qt, min_qt, max_qt : quorum_fraction * quorum_fraction * quorum_fr
   else qt
 
 [@inline]
-// Returns the 'cycle' number of which the given 'stage' number is part of.
-let stage_to_cycle (p: nat): nat = (p + 1n) / 2n
+let period_to_cycle (p: nat): nat = (p + 1n) / 2n
```
[the remark this answers is not in the document]
Comment (sras) [reply src/proposal/quorum_threshold.mligo]: Yes, restored.

File: src/proposal.mligo [hunk 3]
```diff
@@ -250,36 +251,36 @@ let do_total_vote_meet_quorum_threshold (proposal, store: proposal * storage): b
   let reached_quorum = (votes_placed * quorum_denominator) / total_supply in
   (reached_quorum >= proposal.quorum_threshold.numerator)
 
-// Delete a proposal from 'proposal_key_list_sort_by_level'
+// Delete a proposal from 'sProposalKeyListSortByDate'
```
[the remark this answers is not in the document]
Comment (sras) [reply src/proposal.mligo]: Changed

File: src/proposal.mligo [hunk 4]
```diff
@@ -23,16 +23,16 @@ let check_if_proposal_exist (proposal_key, store : proposal_key * storage): prop
...
+let get_current_stage_num(start_time, period : timestamp * period) : nat =
+  match is_nat((Tezos.now - start_time) : int) with
+  | Some (elapsed_time) -> elapsed_time/period.length
   | None -> ([%Michelson ({| { FAILWITH } |} : string * unit -> nat)]
       ("BAD_STATE", ()))
 
 [@inline]
-let ensure_proposal_voting_stage (proposal, voting_period, store : proposal * period * storage): storage =
-  let current_stage = get_current_stage_num(store.start_level, voting_period) in
+let ensure_proposal_stage (proposal, period, store : proposal * period * storage): storage =
```
[the remark this answers is not in the document]
Comment (sras) [reply src/proposal.mligo]: Changed

**Pull request merged**